packages/loopring_v3/contracts/impl/ExchangeV3.sol

```diff
@@ -671,42 +549,106 @@ contract ExchangeV3 is IExchangeV3
     }
 
     function withdrawFromApprovedWithdrawal(
-        uint blockIdx,
-        uint slotIdx
+        address owner,
+        address token
         )
         external
         nonReentrant
     {
-        require(blockIdx < state.blocks.length, "INVALID_BLOCK_IDX");
-        ExchangeData.Block storage withdrawBlock = state.blocks[blockIdx];
         state.withdrawFromApprovedWithdrawal(
-            blockIdx,
-            withdrawBlock,
-            slotIdx,
-            false
+            owner,
+            token
         );
     }
 
-    function withdrawBlockFee(
-        uint blockIdx,
-        address payable feeRecipient
+    function getAmountWithdrawable(
+        address owner,
+        address token
+        )
+        external
+        view
+        returns (uint)
+    {
+        uint24 accountID = (owner != address(0)) ? state.getAccountID(owner) : 0;
+        uint16 tokenID = state.getTokenID(token);
+        return state.amountWithdrawable[accountID][tokenID];
+    }
+
+    // -- Agents --
+    function authorizeAgents(
+        address   owner,
+        address[] calldata agents,
+        bool[]    calldata authorized
+        )
+        external
+        onlyAgentFor(owner)
+    {
+        require(agents.length == authorized.length, "INVALID_DATA");
+        for (uint i = 0; i < agents.length; i++) {
+            state.agent[owner][agents[i]] = authorized[i];
+            emit AgentAuthorized(owner, agents[i], authorized[i]);
+        }
+    }
+
+    function isAgent(address owner, address agent)
+        public
+        view
+        returns (bool)
+    {
+        return owner == agent || state.agent[owner][agent];
+    }
+
+    function approveConditionalTransfer(
+        address from,
```

Comment:
also change 'from' to 'owner'

Reply:
Same explanation here as for the deposit parameter naming.